docs/source/tutorials/token_classification/entity_recognition_training.ipynb

```diff
@@ -0,0 +1,737 @@
+    "entity_map = {entity: i for i, entity in enumerate(entities)} \n",
+    "\n",
+    "def readfile(filepath, sep=' '): \n",
+    "    lines = open(filepath)\n",
+    "    \n",
+    "    data, sentence, label = [], [], []\n",
+    "    for line in lines:\n",
+    "        if len(line) == 0 or line.startswith('-DOCSTART') or line[0] == '\\n':\n",
+    "            if len(sentence) > 0:\n",
+    "                data.append((sentence, label))\n",
+    "                sentence, label = [], []\n",
+    "            continue\n",
+    "        splits = line.split(sep) \n",
+    "        word = splits[0]\n",
+    "        if len(word) > 0 and word[0].isalpha() and word.isupper():\n",
+    "            word = word[0] + word[1:].lower()\n",
+    "        sentence.append(word)\n",
+    "        label.append(entity_map[splits[-1][:-1]])\n",
+    "\n",
+    "    if len(sentence) > 0:\n",
+    "        data.append((sentence, label))\n",
+    "        \n",
+    "    given_words = [d[0] for d in data] \n",
+    "    given_labels = [d[1] for d in data] \n",
+    "    \n",
+    "    return given_words, given_labels \n",
+    "\n",
+    "given_words, given_labels = [], [] \n",
+    "\n",
+    "for filepath in filepaths: \n",
+    "    words, labels = readfile(filepath) \n",
+    "    given_words.extend(words) \n",
+    "    given_labels.extend(labels)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "12148277",
+   "metadata": {},
+   "source": [
+    "`given_words` and `given_labels` are in nested list format. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "c1d505f4",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Word\t\tLabel\tEntity\n",
+      "-------------------------------\n",
+      "Eu              5\tB-ORG     \n",
+      "rejects         0\tO         \n",
+      "German          1\tB-MISC    \n",
+      "call            0\tO         \n",
+      "to              0\tO         \n",
+      "boycott         0\tO         \n",
+      "British         1\tB-MISC    \n",
+      "lamb            0\tO         \n",
+      ".               0\tO         \n"
+     ]
+    }
+   ],
+   "source": [
+    "i = 0 \n",
+    "\n",
+    "print('Word\\t\\tLabel\\tEntity\\n-------------------------------') \n",
+    "for word, label in zip(given_words[i], given_labels[i]): \n",
+    "    print('{:14s}{:3d}\\t{:10s}'.format(word, label, entities[label])) "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "16f8639b",
+   "metadata": {},
+   "source": [
+    "Next, obtain the sentences with some minor pre-processing for readability. Sentences that contain `#` or have length shorter than or equal to one character are removed. The first condition is to ensure that the symbol does not get confused with the same symbol representing subwords after the sentence is being tokenized. "
```

Comment:
Can you try to more clearly explain the special significance of the `#` character?